[scroll-animations] WPT test scroll-animations/view-timelines/animation-events.html is a flaky failure

graouts opened the pull request:
#### 2f89cb3ece5bf30312d87035082caae3ee17cfb5
<pre>
[scroll-animations] WPT test scroll-animations/view-timelines/animation-events.html is a flaky failure
<a href="https://bugs.webkit.org/show_bug.cgi?id=284299">https://bugs.webkit.org/show_bug.cgi?id=284299</a>

Reviewed by Anne van Kesteren.

The Web Animations spec says that pending animation events should be enqueued per document, and we
made the choice years ago to host the event queue on the `DocumentTimeline` created for each document.

When document timelines were the only timeline types, running the &quot;update animation and send events&quot;
procedure [0] would only ever run if the document timeline needed an update. When we started supporting
multiple types of timelines, this meant we could run that procedure when a scroll or view timeline
needed an update but the document timeline did not.

As such, we change the step where we process events in the &quot;update animation and send events&quot; procedure
to not only look for a document timeline among the timelines needing an update, but always looking at
the document timeline created for the document.

A better fix would be to stop enqueuing events on `DocumentTimeline` and instead on
`AnimationTimelinesController` but the given the scope of that potential change and how simple
this fix is, it seems like a better solution for the time being.

[0] <a href="https://drafts.csswg.org/web-animations-1/#update-animations-and-send-events">https://drafts.csswg.org/web-animations-1/#update-animations-and-send-events</a>

* LayoutTests/TestExpectations:
* Source/WebCore/animation/AnimationTimelinesController.cpp:
(WebCore::AnimationTimelinesController::updateAnimationsAndSendEvents):

Canonical link: <a href="https://commits.webkit.org/294068@main">https://commits.webkit.org/294068@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4b9777bf4ab2594e2f3c94f0b69f2aa348df2d1d

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/100745 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/20397 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/10696 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/105882 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/51334 "Failed to checkout and rebase branch from PR 44468") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/102786 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/20706 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/28871 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/105882 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/59/builds/51334 "Failed to checkout and rebase branch from PR 44468") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/103752 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/15909 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/90999 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/105882 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/15718 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/9007 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/50709 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/85621 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/9082 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/108237 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/27863 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/20464 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/108237 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/28226 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/87201 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/108237 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/126/builds/29921 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/7649 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/21867 "Failed to checkout and rebase branch from PR 44468") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/16384 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/27798 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/33053 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/27609 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/30927 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/29167 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk reviewed (approved):
Maybe consider a FIXME comment so people don't have to look at git blame to find the more ideal architecture and why things are this way.

graouts committed:
[scroll-animations] WPT test scroll-animations/view-timelines/animation-events.html is a flaky failure
https://bugs.webkit.org/show_bug.cgi?id=284299

Reviewed by Anne van Kesteren.

The Web Animations spec says that pending animation events should be enqueued per document, and we
made the choice years ago to host the event queue on the `DocumentTimeline` created for each document.

When document timelines were the only timeline types, running the "update animation and send events"
procedure [0] would only ever run if the document timeline needed an update. When we started supporting
multiple types of timelines, this meant we could run that procedure when a scroll or view timeline
needed an update but the document timeline did not.

As such, we change the step where we process events in the "update animation and send events" procedure
to not only look for a document timeline among the timelines needing an update, but always looking at
the document timeline created for the document.

A better fix would be to stop enqueuing events on `DocumentTimeline` and instead on
`AnimationTimelinesController` but the given the scope of that potential change and how simple
this fix is, it seems like a better solution for the time being.

[0] https://drafts.csswg.org/web-animations-1/#update-animations-and-send-events

* LayoutTests/TestExpectations:
* Source/WebCore/animation/AnimationTimelinesController.cpp:
(WebCore::AnimationTimelinesController::updateAnimationsAndSendEvents):

Canonical link: https://commits.webkit.org/294068@main
